lib/secrets.rb

```diff
@@ -65,6 +71,25 @@ def http_post(url, ssl_verify:, pem:)
     response.body
   end
 
+  def hostip
+    token = File.read(@serviceaccount_dir + '/token')
+    namespace = File.read(@serviceaccount_dir + '/namespace')
+    uri = URI.parse(@api_url + "/api/v1/namespaces/#{namespace}/pods")
+    req = Net::HTTP::Get.new(uri.path)
+    req.add_field("Authorization", "Bearer #{token}")
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.use_ssl = (uri.scheme == 'https')
```

Comment:
I was hoping for:

```Ruby
def http_get(url, headers: {}, ca_cert: nil)
```

so the 'http get logic' is separated from the 'vault request logic'